adls-plugins/src/main/java/io/cdap/plugin/sink/ADLSBatchSink.java

```diff
@@ -226,12 +244,28 @@ protected void validate() {
       properties.put("fs.adl.impl", "org.apache.hadoop.fs.adl.AdlFileSystem");
       properties.put("fs.AbstractFileSystem.adl.impl", "org.apache.hadoop.fs.adl.Adl");
       properties.put("dfs.adls.oauth2.access.token.provider.type", "ClientCredential");
-      properties.put("dfs.adls.oauth2.refresh.url", refreshTokenURL);
-      properties.put("dfs.adls.oauth2.client.id", clientId);
-      properties.put("dfs.adls.oauth2.credential", credentials);
+      if (keyVaultUrl != null && !keyVaultUrl.isEmpty()) {
+        Map<String, String> credentials = AzureClientSecretService.getADLSSecretsUsingJceksAndKV(keyVaultUrl, getKvKeyNamesMap(kvKeyNames), properties);
+        properties.put("dfs.adls.oauth2.refresh.url", credentials.get("Refresh Token Url Key-Name"));
+        properties.put("dfs.adls.oauth2.client.id", credentials.get("Client-Id Key-Name"));
+        properties.put("dfs.adls.oauth2.credential", credentials.get("Client Credential Key-Name"));
+      } else {
+        properties.put("dfs.adls.oauth2.refresh.url", refreshTokenURL);
+        properties.put("dfs.adls.oauth2.client.id", clientId);
+        properties.put("dfs.adls.oauth2.credential", credentials);
+      }
       return properties;
     }
 
+    protected HashMap<String, String> getKvKeyNamesMap(String kvKeyNames) {
+       HashMap<String, String> credMap = new HashMap<String, String>();
+       String[] keypairs = kvKeyNames.split(",");
+       for (String k : keypairs) {
+         credMap.put(k.split(":")[1], k.split(":")[0]);
```

Comment:
split once and re-use the resulting array.

Also, are these special characters (',' and ':') allowed in keyvault key names? If so there will be issues with this parsing.